test/e2e/generate-multiversion.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+set -euo pipefail
+
+ARGS=$@
+FLAGS=${ARGS:=-g 4 -d networks/nightly}
+
+# This lists all tags that are reachable from the current commit, sorted by
+# increasing semver version.
+SORTED_TAGS=`git tag -l --merge HEAD  --sort="v:refname"`
+
+# Extracts the major and minor version of the most recent -dev tagged version.
+# For example, on the backport branch for v0.32, the most recent dev tag will contain v0.32.0-dev.
+# v0.32 is then extracted from this tag.
+BRANCH_MINOR_VERSION=`echo "$SORTED_TAGS" | grep '^v[0-9]\+\.[0-9]\+\.[0-9]\+\(-dev[0-9]*\)$' | tail -n 1 | sed -n 's/^\(v[0-9]\+\.[0-9]\+\).*/\1/p'`
+LATEST_RELEASE_VERSION=`echo "$SORTED_TAGS" | grep  '^v[0-9]\+\.[0-9]\+\.[0-9]\+$' | tail -n 1`
+
+# Check to see if the major and minor version of the latest release version are
+# the same as those of the current branch. This ensures that the multi-version testing
+# is only run on branches with a tagged release of the same major and minor version
+# and not on branches that do not yet have releases such as new backport branches
+# and the main branch.
+if [ `echo "$LATEST_RELEASE_VERSION" | grep "^$BRANCH_MINOR_VERSION"` ]; then
```

Comment:
On `main`, for me this evaluates to:

```
v0.34.20
```

Is that intended?

Reply:
I see the issue here, I had an old tag of the old style for v0.38 that looks to not ended up being used (perhaps deleted?). I'm not sure where this tag came from, but pulling the repo afresh shows that that tag is not present and that instead the `dev-` tag is used. I've switched the logic here to first check for the `dev-` tag style before checking for the `<version>-dev` style, which should alleviate this issue as well as the comment below.